docs/architecture/venice_write_path.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: default
+title: Venice Write Path
+parent: Venice Architecture
+permalink: /docs/architecture/venice_write_path
+---
+
+# Introduction
+
+As a high performance derived data storage platform, Venice is designed to ingest large amount of writes from a variety 
+of sources while providing low latency read access to the data. These characteristics are achieved by a combination of
+design choices and optimizations in the write path. This document describes some internal details of the Venice write path. 
+
+Note that this page is still under construction and only provides glimpses of the Venice writing path. More details will 
+be added in the future.
+
+### Terminology
+
+| Name  | Description                                                                                                 |
+|-------|-------------------------------------------------------------------------------------------------------------|
+| VT    | Version Topic. The version topic is a kafka topic where holds all the data for a given store.               |
+| RT    | Real time Topic. A real time topic is also a kafka topic but it only contains data from nearline producers. |
+| ZK    | Apache ZooKeeper. It's used in multiple Venice components to persist Venice metadata.                       |
+| Helix | Apache Helix. It's used in Venice to manage the cluster and assign partitions to different servers.         |
+| EV    | Helix External View. Venice uses it to represent the state of a server in a leader/follower mode.           |
+| CV    | Helix Customized View. Venice uses it to represent the ingestion state of a server for a given store.       |
+
+### Venice Store Ingestion on server side
+
+In Venice, a table or the dataset is called a store and each store has multiple partitions. Helix assign leaders and
+followers status on each partition to different servers. When assignments are done, each server will create multiple
+`StoreIngestionTask`s to handle ingestion for the partitions it's assigned to. The diagram belows describes the data flow of how a 
+store ingestion task's created and ingest data on the server side.
+
+![Server Ingestion Diagram](../assets/images/server_ingestion.svg)
+
+### Venice Store Ingestion report 
+
+When ingestion is in different stages, the server reports the ingestion status. The ingestion signal is propagated to 
+various downstream for different business logics. The diagram below describes the signal flow of the ingestion report.
+
+![Server Ingestion Report Diagram](../assets/images/server_ingestion_report.svg)
```

Comment:
This second diagram is also accurate, I believe. Although I seem to recall that @xunyin8 recently deleted the code which updates the meta store, since FC is using a different mechanism now... If that is true, maybe that part could be deleted in the diagram as well, to simplify things slightly.

Also, I wonder if there is a way in our docs theme to make images pop out in full size when clicking on them... I managed to do it by right-clicking and loading just the image in another browser tab, but it may be a good usability improvement to just be able to do it with a normal click, since these diagrams are pretty detailed and benefit from being seen in full resolution. It's totally fine to leave that for later though.

Reply:
I think you are right. It got removed by #720. Nice to know and clean it up!

The image pop out seems to be a UI-related stuff with the Github page them [just the docs](https://github.com/just-the-docs/just-the-docs). We can revisit it later.